podcasts/End of Year/Stories/TopCategoriesStory/TopListenedCategoriesStory.swift

```diff
@@ -8,47 +8,77 @@ struct TopListenedCategoriesStory: ShareableStory {
 
     let listenedCategories: [ListenedCategory]
 
-    let contrastColor: CategoriesContrastingColors
-
     init(listenedCategories: [ListenedCategory]) {
         self.listenedCategories = listenedCategories
-        self.contrastColor = CategoriesContrastingColors(podcast: listenedCategories.reversed()[0].mostListenedPodcast)
     }
 
     var body: some View {
         GeometryReader { geometry in
             PodcastCoverContainer(geometry: geometry) {
-                let headerSpacing = geometry.size.height * 0.054
+                StoryLabelContainer(geometry: geometry) {
+                    let mostListenedCategory = listenedCategories.first?.categoryTitle.localized ?? ""
+                    let listenedTime = listenedCategories.first?.totalPlayedTime.storyTimeDescription ?? ""
+                    let listenedEpisodes = "\(listenedCategories.first?.numberOfEpisodes ?? 0)"
+                    StoryLabel(L10n.eoyStoryTopCategoriesTitle(mostListenedCategory), for: .title, geometry: geometry)
+                    StoryLabel(L10n.eoyStoryTopCategoriesSubtitle(listenedEpisodes, listenedTime), for: .subtitle, color: Color(hex: "8F97A4"), geometry: geometry)
+                }
 
-                StoryLabel(L10n.eoyStoryTopCategories, for: .title2)
-                    .opacity(0.8)
+                let headerSpacing = geometry.size.height * 0.054
 
-                HStack(alignment: .bottom, spacing: 25) {
-                    ForEach([1, 0, 2], id: \.self) {
-                        pillar($0, size: geometry.size)
+                VStack(alignment: .leading, spacing: geometry.size.height * 0.03) {
+                    ForEach(0...min(listenedCategories.count, 3), id: \.self) { index in
```

Comment:
If I have less than 5 categories, I see the next number but its blank.

<img src="https://github.com/Automattic/pocket-casts-ios/assets/793774/ea35d91c-b0c4-4c54-9ff2-ecf25a5a73d4" width="320" />

```suggestion
                    ForEach(0...<listenedCategories.count, id: \.self) { index in
```